tyro/_instantiators.py

```diff
@@ -93,6 +102,47 @@ class UnsupportedTypeAnnotationError(Exception):
 )
 
 
+def _has_parsable_signature(typ: Union[Callable, TypeForm[Any]]) -> bool:
+    try:
+        inspect.signature(typ)
+    except Exception:
```

Comment:
Maybe still only check for `ValueError`?

Reply:
What's the reason for broadening it to `Exception`?

Reply:
No reason, just looked at the source for `inspect.signature` and it'll only raise `ValueError` so I changed this try block.